Add H47 PCB to QMK

huytbt opened the pull request:
Keyboard with 47-keys layout with block arrow

## Description

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Ordered wrongly layout in info.json
* Update License to GPL-2.0-or-later
* Update readme

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

zvecr commented:
> valid license headers on all *.c and *.h source files
GPL2/GPL3 recommended for consistency
an example GPL2+ license header may be copied and modified from the bottom of this document
other licenses are permitted, however they must be GPL-compatible and must allow for redistribution. Using a different license will almost certainly delay a PR getting merged.

sigprof commented on keyboards/huytbt/h47/info.json:
```diff
@@ -0,0 +1,58 @@
+        { "label": "J", "x": 7.25, "y": 1 },
+        { "label": "K", "x": 8.25, "y": 1 },
+        { "label": "L", "x": 9.25, "y": 1 },
+        { "label": "Enter", "x": 11.25, "y": 1, "w": 1.75 },
+        { "label": "PG DN", "x": 10.25, "y": 1 },
```
The order of these entries looks wrong — it needs to match the order of arguments in the layout macro, and that order is also intended to match the physical key order.

To test this file, you can open https://config.qmk.fm/, press Ctrl+Shift+I, upload the file, then enable the “Fast Input” option and check in which order the keys are filled when you press something.

huytbt replied:
Thank @sigprof very much. My mistake. I fixed

sigprof commented on keyboards/huytbt/h47/info.json:
```diff
@@ -0,0 +1,58 @@
+        { "label": "B", "x": 5.75, "y": 2 },
+        { "label": "N", "x": 6.75, "y": 2 },
+        { "label": "M", "x": 7.75, "y": 2 },
+        { "label": "/", "x": 10.75, "y": 2 },
+        { "label": "Shift", "x": 11.75, "y": 2, "w": 1.25 },
+        { "label": "Up", "x": 8.75, "y": 2 },
+        { "label": "Fn3", "x": 9.75, "y": 2 },
```
This part also seems to be ordered wrongly.

huytbt replied:
Thank @sigprof very much. My mistake. I fixed

sigprof reviewed (changes requested):
Also the layout definition in `info.json` seems to have some errors.

huytbt commented:
> valid license headers on all *.c and *.h source files
> GPL2/GPL3 recommended for consistency
> an example GPL2+ license header may be copied and modified from the bottom of this document
> other licenses are permitted, however they must be GPL-compatible and must allow for redistribution. Using a different license will almost certainly delay a PR getting merged.

I have updated license to GPL-2.0-or-later. Thanks

drashna commented on keyboards/huytbt/h47/config.h:
```diff
@@ -0,0 +1,56 @@
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4859 // HY
+#define PRODUCT_ID      0x0001
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    HUYTBT
+#define PRODUCT         H47
```
```suggestion
```

drashna commented on keyboards/huytbt/h47/info.json:
```diff
@@ -0,0 +1,58 @@
+{
+  "keyboard_name": "H47",
+  "url": "",
+  "maintainer": "huytbt",
```
```suggestion
  "maintainer": "huytbt",
  "usb": {
      "vid": "0x4859",
      "pid": "0x0001",
      "device_version": "0.0.1"
  }
```
